db/fetcher/fetcher.go

```diff
@@ -514,14 +522,49 @@ func (df *DocumentFetcher) processKV(kv *keyValue) error {
 
 	df.execInfo.FieldsFetched++
 
-	df.doc.Properties[fieldDesc] = property
+	df.doc.properties[fieldDesc] = property
 
 	return nil
 }
 
 // FetchNext returns a raw binary encoded document. It iterates over all the relevant
 // keypairs from the underlying store and constructs the document.
 func (df *DocumentFetcher) FetchNext(ctx context.Context) (EncodedDocument, ExecInfo, error) {
```

Comment:
> Wasn't part of this refactor to make it also match the Enumerable interface?

We for sure spoke a bit about that, and I started to look into it, but it felt slightly forced (due to a slight mismatch of funcs like Close and Reset, Init, Start, etc) , and had much less value IMO.  This PR currently makes things much better, and cheaper for myself and Islam to build upon, changing to `Enumerable` would not speed up the Lens work, and I doubt it would help Islam.

We can look at making it `Enumerable` later I think, for now it is effort I would rather not spend, and would likely have very little impact on our productivity in the medium term.

> we can likely move the ExecInfo off the main Fetch method (it feels a little weird here anyway) and have a dedicated method to get the info instead.

I also thought about doing that too, but it is not a new problem, and is pretty common to have stats like this outputted from funcs like this as return values.  It is also extra effort now that does not benefit us anytime soon as far as I can see, and would only slow down the merge of changes that do benefit us right now.

Reply:
> We for sure spoke a bit about that, and I started to look into it, but it felt slightly forced (due to a slight mismatch of funcs like Close and Reset, Init, Start, etc)

OK no problem. Just remembered talking about it and wanted to make sure it was addressed (regardless of in favor or against).

> I also thought about doing that too, but it is not a new problem, and is pretty common to have stats like this outputted from funcs like this

Correct, this isnt a new problem, and not really all that important. The comment was only in relation to *if* we moved forward with the enumerable interface matching.